core/src/main/java/org/testcontainers/core/CreateContainerCmdModifier.java

```diff
@@ -0,0 +1,15 @@
+package org.testcontainers.core;
+
+import com.github.dockerjava.api.command.CreateContainerCmd;
+
+import java.util.function.Function;
+
+/**
+ * Callback interface that can be used to customize a {@link CreateContainerCmd}.
+ */
+public interface CreateContainerCmdModifier {
+    /**
+     * Callback to modify a {@link CreateContainerCmd} instance.
+     */
+    Function<CreateContainerCmd, CreateContainerCmd> modify();
```

Comment:
It looks like my `Function` proposal was confusing, sorry 😅 I meant this signature:
```suggestion
    CreateContainerCmd modify(CreateContainerCmd cmd);
```

Reply:
I was also thinking about it :D. PR updated